### PR TITLE
(fix): MotherDuck config should set SaaS mode at the end

### PR DESCRIPTION
**Pull request by guenp**
- make sure to set `saas_mode` _after_ the MD token is set
- generalize how config settings are parsed from input
- add tests to confirm that SaaS mode also works when I attach a MD database
- pin DuckDB to 1.1.1 for MD tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -222,7 +222,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.9']
+        python-version: ['3.11']
```
Comment (jwills) [inline .github/workflows/main.yml:225]: Doing a separate PR to update all of these now that 3.8 is basically at EOL

File: tests/functional/plugins/motherduck/test_motherduck_attach.py [hunk 2]
```diff
@@ -93,17 +93,17 @@ def run_dbt_scope(self, project, database_name):
         # CREATE DATABASE does not work with SaaS mode on duckdb 1.0.0
         # This will be fixed in duckdb 1.1.0
```
Comment (jwills) [inline tests/functional/plugins/motherduck/test_motherduck_attach.py:94]: Really 1.1.1?

File: tests/functional/plugins/motherduck/test_motherduck_saas_mode.py [hunk 3]
```diff
@@ -89,16 +89,97 @@ def run_dbt_scope(self, project, database_name):
         # CREATE DATABASE does not work with SaaS mode on duckdb 1.0.0
         # This will be fixed in duckdb 1.1.0
```
Comment (jwills) [inline tests/functional/plugins/motherduck/test_motherduck_saas_mode.py:90]: same

File: .github/workflows/main.yml (hunk 1 above)
Comment (guenp) [reply .github/workflows/main.yml]: Cool! I've removed this commit

File: tests/functional/plugins/motherduck/test_motherduck_saas_mode.py (hunk 3 above)
Comment (guenp) [reply tests/functional/plugins/motherduck/test_motherduck_saas_mode.py]: indeed!

File: tests/functional/plugins/motherduck/test_motherduck_attach.py (hunk 2 above)
Comment (guenp) [reply tests/functional/plugins/motherduck/test_motherduck_attach.py]: This actually doesn't need to go here so I removed it!

**Review by jwills — APPROVED** [review]

**Pull request merged**